tools/adventure-pack/goodies/typescript/traverseLevelOrder/index.ts

```diff
@@ -0,0 +1,23 @@
+export function* traverseLevelOrder<
+  T extends { left?: T | null | undefined; right?: T | null | undefined },
+>(root: T | null | undefined): Generator<T[], void, void> {
+  if (!root) {
+    yield [];
```

Comment:
If there's no root, then there are no levels -- so I think this should just be a return?

Reply:
ack!